libs/community/langchain_community/utilities/you.py

```diff
@@ -94,6 +113,74 @@ def validate_environment(cls, values: Dict) -> Dict:
 
         return values
 
+    @root_validator
+    def warn_if_set_fields_have_no_effect(cls, values) -> Dict:
+        if values["endpoint_type"] != "news":
+            news_api_fields = ("search_lang", "ui_lang", "spellcheck")
+            for field in news_api_fields:
+                if values[field]:
+                    warnings.warn(
+                        (
+                            f"News API-specific field '{field}' is set but "
+                            f"`endpoint_type=\"{values['endpoint_type']}\"`. "
+                            "This will have no effect."
+                        ),
+                        UserWarning,
+                    )
+        if values["endpoint_type"] not in ("search", "snippet"):
+            if values["n_snippets_per_hit"]:
+                warnings.warn(
+                    (
+                        "Field 'n_snippets_per_hit' only has effect on "
+                        '`endpoint_type="search"`.'
+                    ),
+                    UserWarning,
+                )
+        return values
+
+    @root_validator
+    def warn_if_deprecated_endpoints_are_used(cls, values) -> Dict:
+        if values["endpoint_type"] == "snippets":
+            warnings.warn(
+                (
+                    f"`endpoint_type=\"{values['endpoint_type']}\"` is deprecated. "
+                    'Use `endpoint_type="search"` instead.'
+                ),
+                DeprecationWarning,
+            )
+        return values
+
+    def _generate_params(self, query: str, **kwargs) -> Dict:
+        """
+        Parse parameters required for different You.com APIs.
+
+        Args:
+            query: The query to search for.
+        """
+        params = {
+            "safesearch": self.safesearch,
+            "country": self.country,
+            **kwargs,
+        }
+
+        # Add endpoint-specific params
+        if self.endpoint_type in ("search", "snippet"):
+            params.update(
+                query=query,
+                num_web_results=self.num_web_results,
```

Comment:
double-checking that `num_web_results` is not relevant to the news API, as previously that was passed in?

Reply:
The [News API](https://documentation.you.com/api-reference/news) has a `count` parameter in lieu of `num_web_results`. Previously `num_web_results` was passed in but the API didn't complain.